bert/src/pooler.rs

```diff
@@ -43,6 +43,13 @@ impl Embedding<Ix1> {
     pub fn to_bytes(&self) -> Vec<u8> {
         self.iter().flat_map(|value| value.to_le_bytes()).collect()
     }
+
+    pub fn normalized(&self) -> Result<Self, InvalidVectorEncounteredError> {
+        match self.view().l2_norm()? {
+            norm if norm <= 0. => Ok(self.mapv(|_| f32::MAX).into()), // very far away
+            norm => Ok(self.mapv(|x| x / norm).into()),
+        }
```

Comment:
:wrench: this mapping (as you mentioned on slack) seems to be related to some previous handling in the cosine similarity computation. since this function is about normalization, and the metric is handled in another processing step, we should map the edge case to a zero vector instead of a f32::MAX vector, otherwise that would be quite unexpected for normalization. then the cosine similarity/dot product can still act on it later on.

in combination with moving in the parts from the trait impl, i'd suggest sth like:
```suggestion
        let view = self.view();
        let norm = view.dot(&view).sqrt();
        if norm.is_finite() {
            let normalized = if norm <= 0. {
                Array1::zeros(self.len())
            } else {
                &self.0 / norm
            };
            Ok(normalized.into())
        } else {
            Err(InvalidVectorEncounteredError)
        }
```

we can use broadcasting to perform the division. also as a side note: the norm can never be strictly less than zero, neither numerically nor mathematically, we only had `<= 0` instead of `== 0` to avoid float comparison lints.

Reply:
ok will change